tests/test_model_name.py

```diff
@@ -0,0 +1,110 @@
+# TODO: Unit test get_model_name
+from tests.compat import assert_outcomes
+
+
+def test_generic_model_with_custom_name_no_warning(testdir):
+    testdir.makepyfile(
+        """
+    from factory import Factory
+    from pytest_factoryboy import named_model, register
+
+    @register
+    class JSONPayloadFactory(Factory):
+        class Meta:
+            model = named_model(dict, "JSONPayload")
+        foo = "bar"
+
+
+    def test_payload(json_payload: dict):
+        assert isinstance(json_payload, dict)
+        assert json_payload["foo"] == "bar"
+
+    """
+    )
+    result = testdir.runpytest("-Werror")  # Warnings become errors
+    assert_outcomes(result, passed=1)
+
+
+def test_generic_model_name_raises_warning(testdir):
+    testdir.makepyfile(
+        """
+    import builtins
+    from factory import Factory
+    from pytest_factoryboy import register
+
+    @register
+    class JSONPayloadFactory(Factory):
+        class Meta:
+            model = dict
+        foo = "bar"
+
+
+    def test_payload(dict):
+        assert isinstance(dict, builtins.dict)
```

Comment:
NIP: why not simply `dict` like in the test above?

Reply:
because `dict` in this context is the object created by the factory